app/main.py

```diff
@@ -1,2 +1,73 @@
+from typing import Any, Union
+
+from app.point import Point
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.size = 0
+        self.hash_table = [[None]] * 8
+
+    def resize(self) -> list:
+        old_hash_table = self.hash_table
+        self.hash_table = [[None]] * len(old_hash_table) * 2
+        self.size = 0
+        for el in old_hash_table:
+            if el[0] is not None:
+                self.__setitem__(el[0], el[1])
+        return self.hash_table
+
+    def __setitem__(
+            self,
+            key: Union[str, int, tuple, Point],
+            value: Any
+    ) -> list:
+        if self.size / len(self.hash_table) >= 0.625:
+            self.hash_table = self.resize()
+
+        len_table = len(self.hash_table)
+        index = hash(key) % len_table
+        if self.hash_table[index][0] is None:
+            self.hash_table[index] = [key, value]
+            self.size += 1
+            return self.hash_table
```

Comment:
why do we return hash_table? we don't need to return here anything.
this method looks like this `d['key'] = 10`